src/schema/document/owned_value.rs

```diff
@@ -58,6 +58,29 @@ impl AsRef<OwnedValue> for OwnedValue {
     }
 }
 
+impl OwnedValue {
+    /// Returns a u8 discriminant value for the `OwnedValue` variant.
+    ///
+    /// This can be used to sort `OwnedValue` instances by their type.
+    pub fn discriminant_value(&self) -> u8 {
+        match self {
+            OwnedValue::Null => 0,
+            OwnedValue::Str(_) => 1,
+            OwnedValue::PreTokStr(_) => 2,
+            OwnedValue::U64(_) => 3,
```

Comment:
```suggestion
            // It is key to make sure U64, I64, F64 are grouped together in there, otherwise we might be breaking transivity.
            OwnedValue::U64(_) => 3,
```